### ISSUE TITLE
JuNest on Android

### ISSUE THREAD

**Issue by theoden8** [fsquillace/junest#119]

**Comment by fsquillace** [comment]
JuNest has still not fully tested on Android. The script should set the variable ARCH as arm.

Do you know what the command `uname -m` returns?

The correct link for the image is:

    https://dl.dropboxusercontent.com/u/42449030/junest/junest-arm.tar.gz

**Comment by theoden8** [comment]
I didn't know that. Everything fails eventually when trying to do:

    chmod +x junest jchroot

Ehh!

**Comment by fsquillace** [comment]
Did you have chance to run `uname -m`?
I am wondering what you get from that command.

The fact that the operation is not permitted might be because those file are owned by a different user. Can you type:

    whoami
    ls -l junest

**Comment by theoden8** [comment]
uname -m
    armv71

    whoami
    unknown uid 10135

    cd .junest/bin
    ls -l junest
    ----rwxr-x 1 1000 1015 7434 <date> junest 

The folder location is my SD card.

**Comment by fsquillace** [comment]
Ok, the architecture looks ok. The problem seems to due to the fact that your current user (10135) is different from the owner of junest (1000).

Also, ~/.junest is usually the directory where the JuNest image should be located. So, if you are installing JuNest by cloning the repository it would be better to place the junest script elsewhere, for instance ~/junest .

**Comment by fsquillace** [comment]
I am planning to try using JuNest on Android by myself as soon as I get my Android phone working again :) it is getting too slow and I might need to reset it.

**Comment by theoden8** [comment]
I failed to use `wget`, `curl` or even firefox do download the repository. So, I just pulled my microSD and cloned the repo from my computer. As for ~/.junest location, I don't believe I have enough space for it on internal memory.

**Comment by fsquillace** [comment]
That probably means the `wget` and `curl` are not available there. I need to create a tutorial which will explain how to get JuNest in Android.

**Comment by theoden8** [comment]
Not exactly, I did have either through Terminal IDE program. Both just didn't get `SSL_CERTIFICATE` or smth.

**Comment by fsquillace** [comment]
Will work on this for the next months. An interesting reference for a Debian image on Android that might be useful: https://play.google.com/store/apps/details?id=com.cuntubuntu

**Comment by fsquillace** [comment]
Few useful references for creating the app. Can be actually useful once the proof of concept will be completed.

- An interesting Android app that [download the image](https://bitbucket.org/linuxonandroid/complete-linux-installer/src/6271dd9b6b15822b69b129e43d182b3056e28698/Complete%20Linux%20Installer%20v2/src/com/zpwebsites/linuxonandroid/DownloadFile.java?at=master&fileviewer=file-view-default)
- [linuxonandroid](https://play.google.com/store/apps/details?id=com.zpwebsites.linuxonandroid)
- [android development](http://lessons.livecode.com/m/2571/l/80751-how-do-i-become-an-android-developer-on-linux)
- [busybox code](https://github.com/meefik/busybox/blob/master/app/src/main/java/ru/meefik/busybox/MainActivity.java) [busybox app](https://play.google.com/store/apps/details?id=ru.meefik.busybox)

Also, please [join the group](https://groups.google.com/forum/#!forum/junest) if you want to discuss the way the proof of concept should be done.

**Comment by NightMachinery** [comment]
Any updates on running junest on Android, particularly in Termux?
